test/parallel/test-buffer-alloc.js

```diff
@@ -1147,6 +1147,14 @@ assert.throws(() => {
   assert.strictEqual(ubuf.buffer.byteLength, 10);
 }
 
+// utf8Write
+{
+  const ubuf = Buffer.allocUnsafeSlow(2 ** 32);
+  assert.strictEqual(ubuf.utf8Write('a', 2, 2 ** 32), 1);
```

Comment:
`utf8Write` is undocumented, we should be testing `toString('utf8')`

Reply:
FWIW, I believe `buf.write` also reproduces the issue.